pkg/stanza/fileconsumer/file_threadpool.go

```diff
@@ -0,0 +1,206 @@
+package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+type ReaderWrapper struct {
```

Comment:
This package needs improvement otherwise, but we should not export this unless necessary.
```suggestion
type readerWrapper struct {
```